### PR TITLE
Fix matching error node information contents so its specific

### PR DESCRIPTION
**Pull request by jayunit100**
First part of the reasons cleanup  #20064  (fix event error messages)  cc @wojtek-t @piosz

### CODE REVIEW

**Comment by k8s-oncall** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37716)

**Comment by jayunit100** [comment]
ping @davidopp this is the first part, fixing the node error messages .  follow on is also in flight ill ping others for that one

File: plugin/pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 1]
```diff
@@ -125,6 +125,17 @@ func PredicateMetadata(p *v1.Pod, nodeInfo map[string]*schedulercache.NodeInfo)
 	return pm.GetMetadata(p, nodeInfo)
 }
 
+func TestErrorMessageForMissingNode(t *testing.T) {
+	pod := &v1.Pod{}
+	// no node information
+	nodeInfo := &schedulercache.NodeInfo{}
+	detailedNodeMessage := "node not found: &NodeInfo{"
+	_, _, errMsg := PodFitsResources(pod, PredicateMetadata(pod, nil), nodeInfo)
+	if !strings.Contains(fmt.Sprintf("%v", errMsg), detailedNodeMessage) {
```
Comment (davidopp) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: should it be "%v}" to close the open brace from two lines above?

Comment (jayunit100) [inline plugin/pkg/scheduler/algorithm/predicates/predicates_test.go]: it could be but we don't want to test the internal string rep of NodeInfo here...?

**Comment by jayunit100** [comment]
closing as this

**Comment by jayunit100** [comment]
actually, now looking at this, i think the base has changed underneat quite a bit, its still worth changing and easy to re add later.

**Pull request closed without merging**